packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py

```diff
@@ -199,3 +200,25 @@ def test_update_overwrite_data(self):
             fig.to_plotly_json()["data"],
             [{"type": "scatter", "y": [1, 3, 2], "line": {"color": "yellow"}}],
         )
+
+    def test_set_subplots(self):
+        # Test that it works the same as make_subplots for a simple call
+        fig0 = go.Figure()
+        fig0_sp = make_subplots(2, 2)
+        fig0.set_subplots(2, 2)
+        assert fig0.layout == fig0_sp.layout
+        # Test that it accepts the same arguments as make_subplots
+        fig1 = go.Figure()
+        fig1.set_subplots(rows=2, cols=2, horizontal_spacing=0.25, vertical_spacing=0.1)
+        fig1_sp = make_subplots(
+            rows=2, cols=2, horizontal_spacing=0.25, vertical_spacing=0.1
+        )
+        assert fig1.layout == fig1_sp.layout
+        # Test that calling on a figure that already has subplots throws an error.
+        raised = False
+        try:
```

Comment:
I'd recommend using `with pytest.raises` (https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) here as it's a little more concise.

Reply:
Originally I used `pytest.raisesRegex` but it broke the Python2 tests.

Reply:
Does the `match` argument to `pytest.raises` work with Python 2.7? e.g. `pytest.raises(ValueError, match=r".* 123 .*")`.

Reply:
:+1: